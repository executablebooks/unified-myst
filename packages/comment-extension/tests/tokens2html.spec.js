import { micromark } from 'micromark'

import { mystCommentMmarkExt, mystCommentHtmlExt } from '../src/index.js'

const fixtures = [
    '%',
    '% ',
    ' %',
    '% abc',
    '% abc\n',
    '% abc\n ',
    '% abc\nd',
    ' % a\n%b\n %c',
    '% &',
    '- a\n% b',
    '% abc\n% def',
    '% abc\n\n% def',
    '% abc\n%\n% def',
]

describe('Convert source to HTML', () => {
    test.each(fixtures)(`%j`, (content) => {
        const options = {
            extensions: [mystCommentMmarkExt],
            htmlExtensions: [mystCommentHtmlExt],
        }
        const htmlText = micromark(content, options)
        expect(htmlText).toMatchSnapshot()
    })
})
