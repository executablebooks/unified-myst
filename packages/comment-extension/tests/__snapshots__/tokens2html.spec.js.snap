// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert source to HTML " % a\\n%b\\n %c" 1`] = `
"<!-- a
b
c -->"
`;

exports[`Convert source to HTML " %" 1`] = `"<!--  -->"`;

exports[`Convert source to HTML "% " 1`] = `"<!--  -->"`;

exports[`Convert source to HTML "% &" 1`] = `"<!-- &amp;amp; -->"`;

exports[`Convert source to HTML "% abc" 1`] = `"<!-- abc -->"`;

exports[`Convert source to HTML "% abc\\n " 1`] = `
"<!-- abc -->
"
`;

exports[`Convert source to HTML "% abc\\n" 1`] = `
"<!-- abc -->
"
`;

exports[`Convert source to HTML "% abc\\n% def" 1`] = `
"<!-- abc
 def -->"
`;

exports[`Convert source to HTML "% abc\\n%\\n% def" 1`] = `
"<!-- abc

 def -->"
`;

exports[`Convert source to HTML "% abc\\n\\n% def" 1`] = `
"<!-- abc -->
<!-- def -->"
`;

exports[`Convert source to HTML "% abc\\nd" 1`] = `
"<!-- abc -->
<p>d</p>"
`;

exports[`Convert source to HTML "%" 1`] = `"<!--  -->"`;

exports[`Convert source to HTML "> % a\\n> % b" 1`] = `
"<blockquote><!-- a
 b -->
</blockquote>"
`;

exports[`Convert source to HTML "- a\\n% b" 1`] = `
"<ul>
<li>a</li>
</ul>
<!-- b -->"
`;
